ccmbioinfo/lfs
# CARDS-1582: Proms mode: Clinicians can delete only forms that they created

[Jira link](https://phenotips.atlassian.net/browse/CARDS-1582)

This PR causes users in the TrustedUsers group in Proms mode to be unable to delete Forms unless they themselves created it.

This also fixes a bug in the DeleteServlet where deleting any versionable node whose parent is inaccessible to the current user would crash with an Access Denied exception.

**To test**:
- Start CARDS under proms mode with trusted permissions active (`./start_cards.sh --dev -P cards4proms --permissions trusted`)
- Create multiple users from the Admin page (Administration->Users->Create)
- Add them all to the TrustedUsers group (Administration->Groups)
- Create Patients/Visits for testing with the Admin account
- Create Visit Information forms while logged in as the new TrustedUsers
- You should only be allowed to delete that form with the user that created it. The other users should receive an error like below.

![image](https://user-images.githubusercontent.com/4656440/152880759-2354e18a-bc68-456f-9558-f57abcf2ed5d.png)
